### PR TITLE
EVP: Fix evp_keymgmt_util_copy() for to->keymgmt == NULL

### PR DESCRIPTION
**Pull request by levitte**
evp_keymgmt_util_copy() didn't treat the case to->keymgmt correctly.
The proper change is to use from->keymgmt when to->keymgmt is NULL.

Fixes coverity #1462553

### CODE REVIEW

**Comment by levitte** [comment]
Note that this change really doesn't matter *for the moment*, 'cause we're only calling this function when both `to->keymgmt != NULL` and `from->keymgmt != NULL`.  However, when legacy code is gone, I expect that this will be called unconditionally by `EVP_PKEY_copy_parameters()`.

(I really should get the "no-legacy-code" config PR started, that's going to demonstrate what we can expect things to look like in the farther future)

File: crypto/evp/keymgmt_lib.c [hunk 1]
```diff
@@ -369,16 +378,15 @@ int evp_keymgmt_util_copy(EVP_PKEY *to, EVP_PKEY *from, int selection)
         import_data.selection = selection;
 
         if (!evp_keymgmt_export(from->keymgmt, from->keydata, selection,
-                                &try_import, &import_data)) {
-            evp_keymgmt_freedata(to_keymgmt, alloc_keydata);
+                                &try_import, &import_data))
             return 0;
```
Comment (slontis) [inline crypto/evp/keymgmt_lib.c:382]: Memory leak?? evp_keymgmt_freedata(to_keymgmt, alloc_keydata);

Comment (levitte) [reply crypto/evp/keymgmt_lib.c:382]: Ah, thanks

Comment (levitte) [reply crypto/evp/keymgmt_lib.c]: Restored

**Comment by levitte** [comment]
Merged

036ee3706352970a15300b5cd4bf0b2cb370e12a EVP: Fix evp_keymgmt_util_copy() for to->keymgmt == NULL

**Pull request closed without merging**